includes/data-events/class-modal-checkout.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Newspack Data Events Modal Checkout helper.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Data_Events;
+
+use Newspack\Data_Events;
+use WP_Error;
+
+/**
+ * Class to register the Modal_Checkout listeners.
+ */
+final class Modal_Checkout {
+
+	/**
+	 * The name of the action for form submissions
+	 */
+	const FORM_SUBMISSION = 'form_submission_received';
+
+	/**
+	 * The name of the action for form submissions
+	 */
+	const FORM_SUBMISSION_SUCCESS = 'form_submission_success';
+
+	/**
+	 * Initialize the class by registering the listeners.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		Data_Events::register_listener(
+			'newspack_blocks_checkout_button_modal',
+			'modal_checkout_interaction',
+			[ __CLASS__, 'checkout_button_purchase' ]
+		);
+
+		Data_Events::register_listener(
+			'newspack_blocks_donate_block_modal',
+			'modal_checkout_interaction',
+			[ __CLASS__, 'donate_button_purchase' ]
+		);
+
+		Data_Events::register_listener(
+			'woocommerce_checkout_order_created',
+			'modal_checkout_interaction',
+			[ __CLASS__, 'checkout_attempt' ]
+		);
+
+		Data_Events::register_listener(
+			'newspack_blocks_modal_continue',
+			'modal_checkout_interaction',
+			[ __CLASS__, 'modal_pagination' ]
+		);
+	}
+
+	/**
+	 * Returns whether a product is a one time purchase, or recurring and when.
+	 *
+	 * @param string $product_id Product's ID.
+	 */
+	public static function get_purchase_recurrence( $product_id ) {
+		$recurrence = get_post_meta( $product_id, '_subscription_period', true );
+		if ( empty( $recurrence ) ) {
+			$recurrence = 'once';
+		}
+		return $recurrence;
+	}
+
+	/**
+	 * Fires when a reader opens the modal checkout from a checkout button block.
+	 *
+	 * @param array $checkout_button_metadata Information about the purchase.
+	 *
+	 * @return ?array
+	 */
+	public static function checkout_button_purchase( $checkout_button_metadata ) {
+		$metadata = [];
+		foreach ( $checkout_button_metadata as $key => $value ) {
+			$metadata[ $key ] = $value;
+		}
+
+		$data = [
+			'action'      => self::FORM_SUBMISSION_SUCCESS,
+			'action_type' => 'paid_membership',
```

Comment:
This default is a bit misleading for simple product purchases. Or is the intention to ignore non-membership, non-donation purchases altogether (which is not the current behavior).

Also, I have a clarifying question. Are regular subscription products considered `paid_membership`'s as well? Or does this refer to subscriptions that are tied to WC Membership plans only?